saidump/saidump.cpp

```diff
@@ -399,7 +417,170 @@ void dumpGraph(const TableDump& td)
+    }
+
+    std::ofstream outputFile(file_name);
+
+    if (!outputFile.is_open())
+    {
+        std::cerr << "Failed to open the output file " << file_name << std::endl;
+        SWSS_LOG_ERROR("Failed to open the output file %s.", file_name.c_str());
+        return SAI_STATUS_FAILURE;
+    }
+
+    outputFile << content;
+    outputFile.close();
+
+    return SAI_STATUS_SUCCESS;
+}
+
+int dump_from_redis_rdb_file(std::string file_name)
+{
+    std::ifstream input_file(file_name);
+
+    if (!input_file.is_open())
+    {
+        std::cerr << "Failed to open the input file " << file_name << std::endl;
+        SWSS_LOG_ERROR("The file %s does not exist for dumping from redis dump.rdb file.", file_name.c_str());
+        return SAI_STATUS_FAILURE;
+    }
+
+    input_file.seekg(0, std::ios::end);     // Move to the end of the file
+    int64_t file_size = input_file.tellg(); // Get the current position
+
+    if (file_size >= RDB_FILE_MAX_SIZE)
+    {
+        std::cerr << "Get " << file_name.c_str() << "'s size failure or its size " << file_size << "  >= "<< RDB_FILE_MAX_SIZE / 1024 / 1024<<" MB." << std::endl;
+        SWSS_LOG_ERROR("Get %s's size failure or its size %ld  >= %ld MB.", file_name.c_str(), file_size, RDB_FILE_MAX_SIZE / 1024 / 1024);
+
+        // Close the input file
+        input_file.close();
+        return SAI_STATUS_FAILURE;
+    }
+
+    input_file.seekg(0, std::ios::beg);     // Move to the begin of the file
+
+    try
+    {
+        // Parse the JSON data from the file (validation)
+        nlohmann::json jsonData;
+        input_file >> jsonData;
+        input_file.close();
+
+        SWSS_LOG_DEBUG("JSON file is valid.");
+
+        for (json::iterator it = jsonData.begin(); it != jsonData.end(); ++it)
+        {
+            json jj_key = it.key();
+
+            std::string keystr = jj_key;
+            std::string item_name = keystr;
+            size_t pos = keystr.find_first_of(":");
+
+            if (pos != std::string::npos)
+            {
+                if(ASIC_STATE_TABLE != keystr.substr(0, pos))  // filter out non ASIC_STATE
+                {
+                    continue;
+                }
+                item_name = keystr.substr(pos + 1);
+                if (item_name.find(":") != std::string::npos)
+                {
+                    item_name.replace(item_name.find_first_of(":"), 1, " ");
+                }
+            }
+            else
+            {
+                continue;
+            }
+
+            std::cout << item_name << " \r\n";
+
+            json jj = it.value();
+
+            if (!it->is_object())
+            {
+                continue;
+            }
+
+            TableMap map;
+
+            for (json::iterator itt = jj.begin(); itt != jj.end(); ++itt)
+            {
+                if (itt.key() == "NULL")
+                {
+                    continue;
+                }
+                map[itt.key()] = itt.value();
```

Comment:
in current implementation, there can be objects that are created but have no attributes, i dont remember if they are displayed like this:
```
SAI_OBJECT_TYPE ...

or

SAI_OBJECT_TYPE ..
    NULL:NULL

Reply:
An example:
```

SAI_OBJECT_TYPE_BRIDGE oid:0x39000000000024

SAI_OBJECT_TYPE_BRIDGE_PORT oid:0x3a000000000171

SAI_OBJECT_TYPE_BRIDGE_PORT oid:0x3a000000000172

SAI_OBJECT_TYPE_BUFFER_POOL oid:0x180000000007ef
    SAI_BUFFER_POOL_ATTR_SIZE           : 6441610000
    SAI_BUFFER_POOL_ATTR_THRESHOLD_MODE : SAI_BUFFER_POOL_THRESHOLD_MODE_DYNAMIC
    SAI_BUFFER_POOL_ATTR_TYPE           : SAI_BUFFER_POOL_TYPE_BOTH
    SAI_BUFFER_POOL_ATTR_XOFF_SIZE      : 11354112

```